File: client-ios/SerenadaCore/Sources/Call/WebRtcFieldTrials.swift
import Foundation
@preconcurrency import WebRTC

private let serenadaWebRtcFieldTrialsLock = NSLock()
private var serenadaWebRtcFieldTrialsInitialized = false

// Audio RED availability in native libwebrtc is process-wide. Enabling the
// capability here is inert until a session with enableOpusRed=true promotes RED
// ahead of Opus on its audio transceivers.
func initializeSerenadaWebRtcFieldTrialsIfNeeded() {
    serenadaWebRtcFieldTrialsLock.lock()
    defer { serenadaWebRtcFieldTrialsLock.unlock() }
    guard !serenadaWebRtcFieldTrialsInitialized else { return }
    RTCInitFieldTrialDictionary(["WebRTC-Audio-Red-For-Opus": "Enabled"])
    serenadaWebRtcFieldTrialsInitialized = true
}

File: client-android/serenada-core/src/main/java/app/serenada/core/call/WebRtcFieldTrials.kt
package app.serenada.core.call

// Enabling the capability on Serenada's native factory is inert until a session
// with enableOpusRed=true promotes RED ahead of Opus on its audio transceivers.
internal const val SERENADA_WEBRTC_FIELD_TRIALS = "WebRTC-Audio-Red-For-Opus/Enabled/"
